<a href="https://colab.research.google.com/github/iraseel74/mining-project/blob/main/phase1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
import pandas as pd

import numpy as np

import matplotlib.pyplot as plt

df = pd.read_csv('/student-por.csv')

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 649 entries, 0 to 648
Data columns (total 33 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   school      649 non-null    object
 1   sex         649 non-null    object
 2   age         649 non-null    int64 
 3   address     649 non-null    object
 4   famsize     649 non-null    object
 5   Pstatus     649 non-null    object
 6   Medu        649 non-null    int64 
 7   Fedu        649 non-null    int64 
 8   Mjob        649 non-null    object
 9   Fjob        649 non-null    object
 10  reason      649 non-null    object
 11  guardian    649 non-null    object
 12  traveltime  649 non-null    int64 
 13  studytime   649 non-null    int64 
 14  failures    649 non-null    int64 
 15  schoolsup   649 non-null    object
 16  famsup      649 non-null    object
 17  paid        649 non-null    object
 18  activities  649 non-null    object
 19  nursery     649 non-null    object
 20  higher    

goal: Our goal is to predict the weekly alcohol consumption from 1 to 5 based on the features

school - student's school - binary

sex - student's sex - binary

age - student's age - numeric

address - student's home address type - binary

famsize - family size - binary

Pstatus - parent's cohabitation status - binary

Medu - mother's education - numeric

Fedu - father's education - numeric

Mjob - mother's job - nominal

Fjob - father's job - nominal

reason - reason to choose this school - nominal

guardian - student's guardian - nominal

traveltime - home to school travel time - numeric

studytime - weekly study time - numeric

failures - number of past class failures - numeric

schoolsup - extra educational support - binary

famsup - family educational support - binary

paid - extra paid classes within the course subject Math or Portuguese - binary

activities - extra-curricular activities - binary

nursery - attended nursery school - binary

higher - wants to take higher education - binary

internet - Internet access at home - binary

romantic - with a romantic relationship - binary

famrel - quality of family relationships - numeric

freetime - free time after school - numeric

goout - going out with friends - numeric

Dalc - workday alcohol consumption - numeric

Walc - weekend alcohol consumption - numeric

health - current health status - numeric

absences - number of school absences - numeric

These grades are related with the course subject, Math or Portuguese:

G1 - first period grade - numeric

G2 - second period grade - numeric

G3 - final grade - numeric

 number of attributes = 33 , number of object = 648

 source https://www.kaggle.com/datasets/uciml/student-alcohol-consumption

class name : alcohol consumption